### PR TITLE
feat(wallets): add keystore example

### LINKED ISSUE #15: Add KeyStore signer example
See: https://github.com/alloy-rs/alloy/blob/28cab90c70ef2df16571e1ec131aebbba7bc4412/crates/signer-wallet/Cargo.toml#L51 (in the `signer-wallet` crate)

### PR DESCRIPTION
**Pull request by zerosnacks**
Closes #15 

- Adds a `keystore_signer` example of signing using an existing keystore file
- Adds a `create_keystore` example showing how to create a keystore file - this is in line with the existing `sign_message` and `sign_permit_hash` of showing a practical use case

### CODE REVIEW

File: examples/wallets/examples/keystore_signer.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    // Spin up an Anvil node.
+    let anvil = Anvil::new().block_time(1).try_spawn()?;
+
+    // Password to decrypt the keystore file with.
+    let password = "test";
+
+    // Set up a wallet using Alice's keystore file.
+    // The private key belongs to Alice, the first default Anvil account.
+    let wallet =
+        Wallet::decrypt_keystore("./examples/wallets/examples/keystore/alice.json", password)?;
```
Comment (mattsse) [inline examples/wallets/examples/keystore_signer.rs:24]: this needs to use manifest dir + join from env var so this can be run from anywhere

**Review by mattsse — CHANGES_REQUESTED** [review]
smol nits

File: examples/wallets/examples/keystore_signer.rs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        value: Some(U256::from(100)),
+        to: address!("d8dA6BF26964aF9D7eEd9e03E53415D37aA96045").into(),
+        nonce: Some(0),
+        gas_price: Some(U256::from(20e9)),
+        gas: Some(U256::from(21000)),
+        ..Default::default()
+    };
+
+    // Broadcast the transaction and wait for the receipt.
+    let receipt = provider.send_transaction(tx).await?.with_confirmations(1).get_receipt().await?;
```
Comment (onbjerg) [inline examples/wallets/examples/keystore_signer.rs:43]: Is the default not 1 conf? If so, can just leave that out

File: examples/wallets/examples/keystore_signer.rs (hunk 1 above)
Comment (zerosnacks) [reply examples/wallets/examples/keystore_signer.rs]: Added `PathBuf::from(env::var("CARGO_MANIFEST_DIR")?).join("examples/keystore/alice.json")`

This should join from the `Cargo.toml` location in the `examples/wallets/` crate

**Review by mattsse — APPROVED** [review]

**Pull request merged**